ddtrace/ci/internal/writer.py

```diff
@@ -0,0 +1,97 @@
+import os
+from typing import Dict
+from typing import Optional
+from typing import TYPE_CHECKING
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.writer import HTTPWriter
+
+from .encoding import AgentlessEncoderV1
+
+
+if TYPE_CHECKING:
+    from ddtrace.sampler import BasePrioritySampler
+    from ddtrace.sampler import BaseSampler
+    from ddtrace.vendor.dogstatsd import DogStatsd
+
+
+log = get_logger(__name__)
+
+DEFAULT_TIMEOUT = 1.0
+
+
+def get_ci_writer_timeout():
+    # type: () -> float
+    return float(os.getenv("DD_CI_WRITER_TIMEOUT_SECONDS", default=DEFAULT_TIMEOUT))
+
+
+class AgentlessWriter(HTTPWriter):
+    """Writer to the CI Visibility intake endpoint."""
+
+    RETRY_ATTEMPTS = 5
+
+    REQUEST_HTTP_METHOD = "POST"
+
+    ENDPOINT_TEMPLATE = "https://citestcycle-intake.{}"
+
+    def __init__(
+        self,
+        intake_url=None,  # type: Optional[str]
+        sampler=None,  # type: Optional[BaseSampler]
+        priority_sampler=None,  # type: Optional[BasePrioritySampler]
+        processing_interval=5.0,  # type: float
+        # Match the payload size since there is no functionality
+        # to flush dynamically.
+        buffer_size=None,  # type: Optional[int]
+        max_payload_size=None,  # type: Optional[int]
+        timeout=get_ci_writer_timeout(),  # type: float
+        dogstatsd=None,  # type: Optional[DogStatsd]
+        report_metrics=False,  # type: bool
+        sync_mode=True,  # type: bool
+        reuse_connections=None,  # type: Optional[bool]
+        headers=None,  # type: Optional[Dict[str, str]]
```

Comment:
It appears the following parameters are not relevant:

- `sampler`
- `priority_sampler`
- `processing_interval` potentially if there is a non-sync mode
- `dogstatsd`
- `report_metrics`